### PR TITLE
Feature Flags and Rollouts - Incomplete - Do not merge

### PR DESCRIPTION
**Pull request by oakbani**
🖊 Data Models ✔️  
🖊 Data File Parsing ✔️ Unit Testing ❌ 
🖊 Decision Service Logic ❌ 
🖊 Feature Enabled ❌ 
🖊 Feature Variable Accessors ❌

### CODE REVIEW

File: src/Optimizely/Entity/FeatureFlag.php [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+	 */
+	private $_rolloutId;
+
+	/**
+	 * A list of the IDs of the experiments the feature flag is attached to. If there are multiple expeirments,
+	 * they must be in the same mutually exclusive group.
+	 * @var [String]
+	 */
+	private $_experimentIds;
+
```
Comment (msohailhussain) [inline src/Optimizely/Entity/FeatureFlag.php]: need to add variable to key live variable

File: src/Optimizely/Entity/FeatureFlag.php [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+	 * @return String feature flag key
+	 */
+	public function getKey(){
+		return $this->_key;
+	}
+
+	/**
+	 * @param String $key feature flag key
+	 */
+	public function setKey($key){
```
Comment (msohailhussain) [inline src/Optimizely/Entity/FeatureFlag.php]: do we really need any setter method?

File: src/Optimizely/Entity/FeatureVariable.php [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Optimizely\Entity;
+
+class FeatureVariable{
```
Comment (msohailhussain) [inline src/Optimizely/Entity/FeatureVariable.php]: LiveVariable or featurevariable?

File: src/Optimizely/Entity/FeatureVariable.php [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+	/**
+	 * variable containing the string representation of the default value for the feature variable. 
+	 * This is the variable value that should be returned if the user is not bucketed into any variations
+	 * or fails the audience rules.
+	 * @var String
+	 */
+	private $_defaultValue;
+
+
+	public function __construct($id =null, $key = null, $type = null, $defaultValue = null){
```
Comment (msohailhussain) [inline src/Optimizely/Entity/FeatureVariable.php]: fromString method is missing here.

File: src/Optimizely/Entity/VariableUsage.php [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Optimizely\Entity;
+
+class VariableUsage{
```
Comment (msohailhussain) [inline src/Optimizely/Entity/VariableUsage.php]: livevariable usage? in java.

File: src/Optimizely/Entity/FeatureFlag.php (hunk 2 above)
Comment (oakbani) [reply src/Optimizely/Entity/FeatureFlag.php]: I didn't put them first. But they parse json into actual objects using this method by calling setters of each class. So I followed the convention. See here:
https://github.com/msohailhussain/php-sdk/blob/master/src/Optimizely/Utils/ConfigParser.php#L38

File: src/Optimizely/Entity/FeatureVariable.php (hunk 3 above)
Comment (oakbani) [reply src/Optimizely/Entity/FeatureVariable.php]: featurevariable as per requirements doc. Java used livevariable, python only uses variable. I think featurevariable is most appropriate

File: src/Optimizely/Entity/FeatureVariable.php (hunk 4 above)
Comment (oakbani) [reply src/Optimizely/Entity/FeatureVariable.php]: didn't feel a need to add one

**Pull request closed without merging**